lib/puppet-lint/lexer.rb

```diff
@@ -150,7 +150,7 @@ def tokenise(code)
         found = false
 
         KNOWN_TOKENS.each do |type, regex|
-          if value = chunk[regex, 1]
+          if value = chunk[regex, 1].nil? ? 0 : chunk[regex, 1]
```

Comment:
This condition will always pass. `value` will be set to either `0` or `chunk[regex, 1]`.